File: Frontend/src/app/Core/Models/recently-viewed.models.ts
export interface AddRecentlyViewedRequest {
  productId: string;
  userId: string;
}

export interface RecentlyViewedResponse {
  items: RecentlyViewedItem[];
}

export interface RecentlyViewedItem {
  productId: string;
  productName: string;
  price: number;
  imageUrl: string;
  viewedAt: string;
}

File: Backend/eCommerce.Application/DTOs/RecentlyViewed/RecentlyViewedModels.cs
namespace eCommerce.Application.DTOs.RecentlyViewed
{
    public class AddRecentlyViewedRequest
    {
        public Guid ProductId { get; set; }
        public string UserId { get; set; } = string.Empty;
    }

    public class RecentlyViewedResponse
    {
        public List<RecentlyViewedItem> Items { get; set; } = new();
    }

    public class RecentlyViewedItem
    {
        public Guid ProductId { get; set; }
        public string ProductName { get; set; } = string.Empty;
        public decimal Price { get; set; }
        public string ImageUrl { get; set; } = string.Empty;
        public DateTime ViewedAt { get; set; }
    }
}